src/main/java/net/dv8tion/jda/internal/JDAImpl.java

```diff
@@ -829,7 +830,11 @@ public void setAudioSendFactory(IAudioSendFactory factory)
 
     public void setPing(long ping)
     {
+        long oldPing = this.ping;
         this.ping = ping;
+        getEventManager().handle(
+            new GatewayPingEvent(
+                this, oldPing));
```

Comment:
This is technically a side-effect but should be ok.

Reply:
I have proposed a new proxy to avoid exceptions from handle() calls which could cause issues